text/docs-code-contributions.md

```diff
@@ -0,0 +1,69 @@
+- Start Date: 2019-02-07)
+- RFC PR: 
+- Gatsby Issue:
+
+# Summary
+
+Reorganize the contributor documentation to aid in Gatsby core and docs development.
+
+# Motivation
+
+The Gatsby.js docs, while wonderful, are light on coding instructions for contributors. With dedicated sections in the Contributing docs, it will be easier to learn how to clone the repository, where to find certain things like docs or plugins, how to run tests, and insider infrastructure knowledge required to submit a successful pull request on the first try.
+
+A lot of this information already lives on the "How to Contribute" page, but it's buried. It would help discoverability if this information lived on separate pages.
+
+# Detailed design
+
+I propose that we add some pages to the existing contributor documentation and move some content around. Here is an IA diagram that (briefly) reflects the current structure and what I think we should add (open for discussion, of course):
+
+Docs
+- Guides
+- Ecosystem
+- API Reference
+- Releases & Migration
+- Conceptual Guide
+- Behind the Scenes
+- Advanced Tutorials
+
+Contributing
+- Community
+    - Why contribute to Gatsby?
+    - Pair Programming Sessions
+    - Free swag for contributors
+    - Where to participate in the community
+    - How to run a Gatsby workshop*
+    - How to pitch Gatsby*
+- Code of Conduct
+- Gatsby Style Guide
+- How to contribute
+    * Filing & Managing issues
```

Comment:
It's a little funky to combine filing & managing in one doc I think...hence the need to organize by personas, like you mentioned

Reply:
I could make an argument both ways. On the one hand maybe they should be distinct since it's different use cases, on the other hand maybe we don't want to create extra docs...

Reply:
Once I started moving and writing content I decided to keep the two existing pages (`how to file an issue`, `how to label an issue`) and add one new one: `managing issues and pull requests`. That seemed like the most straight forward way to add more content for OSS maintainers and contributors while preserving what we have now.